index c454583..2504886 100644
--- a/autologout.module
+++ b/autologout.module
@@ -341,5 +341,17 @@ function autologout_user_login($account) {
   }

   // Add login time cookie.
-  user_cookie_save(['autologout_login' => \Drupal::time()->getCurrentTime()]);
+  $config = \Drupal::config('autologout.settings');
+  setrawcookie(
+    'Drupal.visitor.autologout_login',
+    rawurlencode(\Drupal::time()->getCurrentTime()),
+    [
+      'expires' => REQUEST_TIME + 31536000,
+      'path' => '/',
+      'samesite' => $config->get('cookie_samesite') ?: 'Lax',
+      'secure' => $config->get('cookie_secure') ?: FALSE,
+      'httponly' => $config->get('cookie_httponly') ?: FALSE,
+    ]
+  );
+
 }
diff --git a/config/install/autologout.settings.yml b/config/install/autologout.settings.yml
index 4d036b3..8690870 100644
--- a/config/install/autologout.settings.yml
+++ b/config/install/autologout.settings.yml
@@ -17,4 +17,7 @@ jstimer_format: '%hours%:%mins%:%secs%'
 jstimer_js_load_option: false
 use_alt_logout_method: false
 use_watchdog: true
-whitelisted_ip_addresses: ''
\ No newline at end of file
+whitelisted_ip_addresses: ''
+cookie_secure: false
+cookie_samesite: 'Lax'
+cookie_httponly: false
diff --git a/config/schema/autologout.schema.yml b/config/schema/autologout.schema.yml
index 1995ab6..202c136 100644
--- a/config/schema/autologout.schema.yml
+++ b/config/schema/autologout.schema.yml
@@ -73,6 +73,15 @@ autologout.settings:
     whitelisted_ip_addresses:
       type: string
       label: 'Whitelisted IP addresses'
+    cookie_secure:
+      type: boolean
+      label: 'Cookie Secure'
+    cookie_samesite:
+      type: string
+      label: 'Cookie SameSite'
+    cookie_httponly:
+      type: boolean
+      label: 'Cookie httpOnly'

 autologout.role.*:
   type: config_object
diff --git a/src/Form/AutologoutSettingsForm.php b/src/Form/AutologoutSettingsForm.php
index d5695d9..b3f2493 100644
--- a/src/Form/AutologoutSettingsForm.php
+++ b/src/Form/AutologoutSettingsForm.php
@@ -284,6 +284,31 @@ class AutologoutSettingsForm extends ConfigFormBase {
         '#description' => $this->t('Change the display of the dynamic timer. Available replacement values are: %day%, %month%, %year%, %dow%, %moy%, %years%, %ydays%, %days%, %hours%, %mins%, and %secs%.'),
       ];
     }
+    $form['cookie_secure'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Cookie Secure'),
+      '#default_value' => $config->get('cookie_secure'),
+      '#description' => $this->t("Whether to set the Secure attribute on the autologout cookie."),
+    ];
+
+    $form['cookie_samesite'] = [
+      '#type' => 'select',
+      '#options' => [
+        'Strict' => $this->t('Strict'),
+        'Lax' => $this->t('Lax'),
+        'None' => $this->t('None'),
+      ],
+      '#title' => $this->t('Cookie SameSite'),
+      '#default_value' => $config->get('cookie_samesite'),
+      '#description' => $this->t("The SameSite attribute to set on the autologout cookie."),
+    ];
+
+    $form['cookie_httponly'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Cookie HttpOnly'),
+      '#default_value' => $config->get('cookie_httponly'),
+      '#description' => $this->t("Whether to set the HttpOnly attribute on the autologout cookie."),
+    ];

     $form['role_container'] = [
       '#type' => 'container',
@@ -471,6 +496,9 @@ class AutologoutSettingsForm extends ConfigFormBase {
       ->set('whitelisted_ip_addresses', $values['whitelisted_ip_addresses'])
       ->set('use_alt_logout_method', $values['use_alt_logout_method'])
       ->set('use_watchdog', $values['use_watchdog'])
+      ->set('cookie_secure', $values['cookie_secure'])
+      ->set('cookie_samesite', $values['cookie_samesite'])
+      ->set('cookie_httponly', $values['cookie_httponly'])
       ->save();

     if (!empty($values['table'])) {
