diff --git a/modules/hdbt_content/hdbt_content.module b/modules/hdbt_content/hdbt_content.module
index df64b6f7..e336f5eb 100644
--- a/modules/hdbt_content/hdbt_content.module
+++ b/modules/hdbt_content/hdbt_content.module
@@ -401,6 +401,7 @@ function _hdbt_content_handle_sidebar_visibility(array &$variables, ContentEntit
     'news_item' => 'node',
     'tpr_unit' => 'tpr_unit',
     'tpr_service' => 'tpr_service',
+    'service' => 'node'
   ];

   // Trigger hook_sidebar_visibility_allowed_entities_alter().
